specs/SystemCursorId.md

```diff
@@ -0,0 +1,122 @@
+# Background
+
+When an app is using composition hosting to host WebView, the current cursor
+image cannot be determined easily by the hosting application. For example, if
+the cursor over the WebView is also over a textbox inside of the WebView, the
+cursor should usually be IDC_IBEAM. And by default, the cursor is IDC_ARROW.
+Currently, we return the what the cursor should be as an HCURSOR object (After
+calling LoadCursor on those IDC values). The hosting application can then use
+the HCURSOR object to change the current cursor to the correct image.
+
+However, Office already has a well established way of using cursors internally
+by using IDC_* values and there is no easy way to convert the HCURSOR object we
+are returning into an IDC value after it's been created.
+
+This new API is to enable hosting applications like Office to get the current
+IDC value of the cursor instead of the HCURSOR as an option.
+
+Note that this is a COM and .NET spec as WinRT uses a separate enum structure to
+identify cursor IDs.
+
+
+# Description
+
+The `SystemCursorId` property will return the current system cursor ID reported
+by the underlying rendering engine for WebView2. It is not meant to return a
+value for any custom cursors, such as those defined by CSS.
+
+It can be used at any time but will only change after a `CursorChanged` event.
+
+Note that developers should generally use the `Cursor` property to support cases
+where there may be custom cursors.
+
+
+# Examples
+
+```cpp
+CHECK_FAILURE(m_compositionController->add_CursorChanged(
+    Callback<ICoreWebView2ExperimentalCursorChangedEventHandler>(
+        [this](ICoreWebView2ExperimentalCompositionController* sender, IUnknown* args)
+            -> HRESULT {
+            HRESULT hr = S_OK;
+            HCURSOR cursor;
+            UINT32 cursorId;
+            wil::com_ptr<ICoreWebView2ExperimentalCompositionController2> compositionController2 =
+                m_controller.query<ICoreWebView2ExperimentalCompositionController2>();
+            CHECK_FAILURE(compositionController2->get_SystemCursorId(&cursorId));
+            cursor = ::LoadCursor(nullptr, MAKEINTRESOURCE(cursorId));
```

Comment:
Presumably should verify that `cursorId != 0`?

Reply:
Yes, good point!